command/agent/template/template.go

```diff
@@ -311,6 +311,15 @@ func newRunnerConfig(sc *ServerConfig, templates ctconfig.TemplateConfigs) (*ctc
 		Enabled:  &enabled,
 	}
 
+	// Sync Consul Template's retry with user set auto-auth initial backoff value.
+	// This is helpful if Auto Auth cannot get a new token and CT is trying to fetch
+	// secrets.
+	if sc.AgentConfig.AutoAuth != nil && sc.AgentConfig.AutoAuth.Method != nil {
+		if sc.AgentConfig.AutoAuth.Method.MinBackoff > 0 {
+			conf.Vault.Retry.Backoff = &sc.AgentConfig.AutoAuth.Method.MinBackoff
```

Comment:
If we decide to go with this, can we add a note in the docs that `min_backoff` also affects backoff behavior for templating retries?

Reply:
If we're push the `min_backoff` to CT, I think that it makes sense to push the `max_backoff` as well. Otherwise CT could get into a state where the `max_backoff` is greater than the min_backoff value.

Note that pushing either value to templating changes the defaults though since they're different:

| Parameter    | Default auto-auth | Default templating
| ----------- | ----------- | ----------- |
| min_backoff      | 1s       | 250ms | 
| max_backoff   | 5min        | 1min |

Related to this (though it can be tackled separately), I wonder if `min_backoff` and `max_backoff` should live outside of the `auto_auth` block if this is the behavior.

Reply:
Sure, no problem.